pkg/resources/nlb_service/datasource_list.go

```diff
@@ -0,0 +1,285 @@
+				},
+			},
+			NLBServiceListAttrNLBID: schema.StringAttribute{
+				Optional: true,
+			},
+			NLBServiceListAttrNLBName: schema.StringAttribute{
+				Optional: true,
+			},
+			NLBServiceListAttrNLBServiceList: schema.ListNestedAttribute{
+				Computed: true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						NLBServiceAttrDescription: schema.StringAttribute{
+							Computed: true,
+						},
+						NLBServiceAttrID: schema.StringAttribute{
+							Computed: true,
+						},
+						NLBServiceAttrHealthcheck: schema.ObjectAttribute{
+							Computed: true,
+							AttributeTypes: map[string]attr.Type{
+								NLBServiceHealthcheckAttrInterval: types.Int64Type,
+								NLBServiceHealthcheckAttrMode:     types.StringType,
+								NLBServiceHealthcheckAttrPort:     types.Int64Type,
+								NLBServiceHealthcheckAttrRetries:  types.Int64Type,
+								NLBServiceHealthcheckAttrTimeout:  types.Int64Type,
+								NLBServiceHealthcheckAttrTLSSNI:   types.StringType,
+								NLBServiceHealthcheckAttrURI:      types.StringType,
+							},
+						},
+						NLBServiceAttrInstancePoolID: schema.StringAttribute{
+							Computed: true,
+						},
+						NLBServiceAttrName: schema.StringAttribute{
+							Computed: true,
+						},
+						NLBServiceAttrPort: schema.Int64Attribute{
+							Computed: true,
+						},
+						NLBServiceAttrProtocol: schema.StringAttribute{
+							Computed: true,
+						},
+						NLBServiceAttrState: schema.StringAttribute{
+							Computed: true,
+						},
+						NLBServiceAttrStrategy: schema.StringAttribute{
+							Computed: true,
+						},
+						NLBServiceAttrTargetPort: schema.Int64Attribute{
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"timeouts": timeouts.Block(ctx, timeouts.Opts{
+				Read: true,
+			}),
+		},
+	}
+}
+
+func (d *NLBServiceListDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data DataSourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	t, diags := data.Timeouts.Read(ctx, config.DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(ctx, t)
+	defer cancel()
+
+	ctx = exoapi.WithEndpoint(ctx, exoapi.NewReqEndpoint(d.env, data.Zone.ValueString()))
+
+	var x string
+	switch {
+	case !data.NLBID.IsNull():
+		x = data.NLBID.ValueString()
+	case !data.NLBName.IsNull():
+		x = data.NLBName.ValueString()
+	default:
+		resp.Diagnostics.AddError(
+			"Either nlb_name or nlb_id must be specified",
+			"",
+		)
+		return
+	}
+
+	nlb, err := d.client.FindNetworkLoadBalancer(ctx, data.Zone.ValueString(), x)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to find Network Load Balancer",
+			err.Error(),
+		)
+		return
+	}
+
+	data.NLBName = types.StringValue(*nlb.Name)
```

Comment:
I just saw that I oversight assigning the NLB ID in case the NLB Name is given, this will result in empty NLB ID field from the state. I'm adding the missing attribution.

Reply:
I've added checks regarding this on https://github.com/exoscale/terraform-provider-exoscale/pull/282/commits/401c33ad69b87ef7f433afc70a5a7281b5c06448